app/form_models/publishers/job_listing/extend_deadline_form.rb

```diff
@@ -0,0 +1,76 @@
+class Publishers::JobListing::ExtendDeadlineForm
+  include ActiveModel::Model
+  include ActiveRecord::AttributeAssignment
+
+  attr_accessor :expires_on, :expiry_time, :starts_on, :starts_asap, :previous_deadline
+
+  validates :expires_on, presence: true, date: true
+  validates :starts_on, date: true, if: proc { starts_on.present? && starts_asap == "0" }
+  validates :expiry_time, inclusion: { in: %w[9:00 12:00 17:00 23:59] }
+
+  validate :expires_at_extended
+  validate :expires_at_in_future
+  validate :starts_on_in_future
+  validate :starts_on_after_expires_at
+  validate :starts_on_and_starts_asap_not_present
+
+  def attributes_to_save
+    {
+      expires_on: expires_on,
+      expires_at: expires_at,
+      starts_on: (starts_on unless starts_asap == "true"),
+      starts_asap: starts_asap,
+    }
+  end
+
+  def expiry_time_options
+    [
+      ["9:00", "Start of the working day (9 am)"],
```

Comment:
should these get surfaced from a locale file somehow?

Reply:
How's that?